src/Transports.Subscriptions.WebSockets/WebsocketWriterStream.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.IO;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GraphQL.Server.Transports.WebSockets
+{
+    public class WebsocketWriterStream : Stream
+    {
+        private readonly WebSocket _webSocket;
+
+        public WebsocketWriterStream(WebSocket webSocket)
+        {
+            _webSocket = webSocket;
+        }
+
+        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            return _webSocket.SendAsync(new ArraySegment<byte>(buffer, offset, count), WebSocketMessageType.Text, false,
+                cancellationToken);
+        }
+
+        public override Task FlushAsync(CancellationToken cancellationToken)
+        {
+            return _webSocket.SendAsync(new ArraySegment<byte>(Array.Empty<byte>()), WebSocketMessageType.Text, true, cancellationToken);
+        }
+
+        public override void Flush()
+        {
+            throw new System.NotImplementedException("Synchronous methods are not supported by WebsocketWriterStream.");
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            throw new System.NotImplementedException();
```

Comment:
Not a big deal, but in this and other methods, `NotSupportedException` would be more appropriate than `NotImplementedException`.